src/error.rs

```diff
@@ -0,0 +1,27 @@
+use serde_json::error::Error as SerdeError;
+use hyper::StatusCode;
+use tungstenite;
+
+error_chain! {
+    foreign_links {
+        Io(std::io::Error);
+        Json(SerdeError);
+        UTF8(std::string::FromUtf8Error);
+    }
+    errors {
+        Status(s: StatusCode) {
+            description("HTTP Status Code")
+            display("HTTP Status: '{}'", s)
+        }
+
+        Limited(s: StatusCode, t: Option<i64>) {
+            description("Reached API Limits")
+            display("{} Retry in: '{:?}'", s, t)
+        }
+
+        Tungstenite(e: tungstenite::Error, t: String) {
+            description("Failed WS")
+            display("{} {}", e, t)
+        }
+    }
+}
```

Comment:
'nother missing newline

Reply:
I will be amending the Error a bit more so I will fix it in the next one. Still need to return "Closed" by WS as Error so it can be handled.